tools/Statistics/CGAL/CGALStatisticsEngine.cpp

```diff
@@ -0,0 +1,76 @@
+/* This file is part of PyMesh. Copyright (c) 2020 by Julien Jerphanion */
+#ifdef WITH_CGAL
+#include "CGALStatisticsEngine.h"
+
+#include <Core/EigenTypedef.h>
+#include <Core/Exception.h>
+#include <CGAL/Polygon_mesh_processing/measure.h>
+#include <CGAL/Polygon_mesh_processing/self_intersections.h>
+
+#include "CGALUtils.h"
+#include "CGALTypes.h"
+#include "statistics_helpers.h"
+
+
+using namespace PyMesh;
+
+void CGALStatisticsEngine::convert_mesh_to_native_format() {
+    Polyhedron polyhedron = CGALUtils::mesh_to_polyhedron<Kernel>(m_vertices, m_faces);
+    CGAL::copy_face_graph(polyhedron, m_mesh);
```

Comment:
Now that it works, add an adaptation to directly convert it to `CGAL::Surface_mesh` without using `CGAL::Polyhedron_3`.